docs/_docs/internals/syntax.md

```diff
@@ -211,6 +211,12 @@ FunArgType        ::=  Type
 FunArgTypes       ::=  FunArgType { ‘,’ FunArgType }
 ParamType         ::=  [‘=>’] ParamValueType
 ParamValueType    ::=  Type [‘*’]                                               PostfixOp(t, "*")
+                    |  IntoType
+                    |  ‘(’ IntoType ‘)’ ‘*’                                     PostfixOp(t, "*")
+IntoType          ::=  [‘into’] IntoTargetType                                  Into(t)
```

Comment:
True, but the recognized language the same either way. It's just the AST construction which is ambiguous. But the grammar does not guarantee that anyway.

The `into` needs to be optional since `IntoType` is also used after `=>` in `IntoTargetTyope`.